### DISCUSSION TITLE
How do I retrieve value of a field in "OnModelBeforeCreate" hook?

### DISCUSSION THREAD

**Discussion by KyungWonPark** [pocketbase/pocketbase#595] [Q&A] [answered]
Hello

I'd like to get value of a field, of certain model
before that record is commited into the database

Let's say I want to get value of a field "price" (type: int), from table "toys"
and do something with it

What should I do here?

```golang
app.OnModelBeforeCreate().Add(func(e *core.ModelEvent) error {
    if e.Model.TableName() == "toys" {
        // print the "price" field of this toy
    }
    
    return nil
})
```

Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
You can do type assertion on the`e.Model` interface:

```go
app.OnModelBeforeCreate().Add(func(e *core.ModelEvent) error {
	if record, ok := e.Model.(*models.Record); ok && record.Collection().Name == "toys" {
		price := record.GetIntDataValue("price")

		log.Println(price)
	}

	return nil
})
```

_footnote: please note that the record accessor will slightly change in v0.8.0 and `GetIntDataValue` will be renamed to just `GetInt`._

**Reply by KyungWonPark** [reply]
Thanks again!! Thank you so much for fast response
